### PR TITLE
Add Linux cwd process column

### PR DESCRIPTION
**Pull request by cgzones**
Related: #344

### CODE REVIEW

File: linux/LinuxProcess.h [hunk 1]
```diff
@@ -19,14 +19,15 @@ in the source distribution for its full text.
...
-#define PROCESS_FLAG_LINUX_SECATTR  0x8000
+#define PROCESS_FLAG_LINUX_IOPRIO   0x00100
+#define PROCESS_FLAG_LINUX_OPENVZ   0x00200
+#define PROCESS_FLAG_LINUX_VSERVER  0x00400
+#define PROCESS_FLAG_LINUX_CGROUP   0x00800
+#define PROCESS_FLAG_LINUX_OOM      0x01000
+#define PROCESS_FLAG_LINUX_SMAPS    0x02000
+#define PROCESS_FLAG_LINUX_CTXT     0x04000
+#define PROCESS_FLAG_LINUX_SECATTR  0x08000
+#define PROCESS_FLAG_LINUX_CWD      0x10000
```
Comment (BenBE) [inline linux/LinuxProcess.h:30]: Cf. #238.

File: linux/LinuxProcessList.c [hunk 2]
```diff
@@ -785,6 +785,27 @@ static void LinuxProcessList_readSecattrData(LinuxProcess* process, const char*
    process->secattr = xStrdup(buffer);
 }
 
+static void LinuxProcessList_readCwd(LinuxProcess* process, const char* dirname, const char* name) {
+   char filename[MAX_NAME + 1];
+   xSnprintf(filename, sizeof(filename), "%s/%s/cwd", dirname, name);
+
+   char pathBuffer[PATH_MAX + 1];
+   ssize_t r = readlink(filename, pathBuffer, sizeof(pathBuffer) - 1);
```
Comment (BenBE) [inline linux/LinuxProcessList.c:793]: Use `readlinkat` via #321.

**Review by BenBE — CHANGES_REQUESTED** [review]
What's the reasoning for renaming `flags` to `processFlags` for this PR?

**Comment by BenBE** [comment]
Checking if all processes in `/proc` have a `cwd` symlink:
```bash
bash -c 'for a in /proc/*/; do [ -e $a/cwd ] || echo $a; done' # as root
```
Returns only non-PID directories for me.

IF you can read the symlink you should always get a non-empty string back. Thus `NULL` might act as a `no perm` marker. Haven't checked for processes running in different FS namespaces though.

**Comment by BenBE** [comment]
Three issues I noticed while testing:

1. Overflows if path is too long (causes mis-alignment).
2. Regularly just shows `/proc/<pid>/fdinfo (deleted)` as cwd
3. `N/A` should be shadowed.

**Comment by cgzones** [comment]
> * Overflows if path is too long (causes mis-alignment).

ideas?

> * Regularly just shows `/proc/<pid>/fdinfo (deleted)` as cwd

resolved

> * `N/A` should be shadowed.

resolved

**Comment by BenBE** [comment]
> > * Overflows if path is too long (causes mis-alignment).
> 
> ideas?
> 
What about `xSnprintf(buffer, n, "%-25.25s", lp->cwd);` ???

**Review by BenBE — APPROVED** [review]

**Pull request merged**